pkg/monitoring/migration/prometheus.go

```diff
@@ -21,133 +21,13 @@ package migration
-		prometheus.GaugeOpts{
-			Name: "kubevirt_migrate_vmi_scheduling_count",
-			Help: "Number of current scheduling migrations.",
-		},
-		migrationsLabels,
-	)
-
-	CurrentRunningMigrations = prometheus.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Name: "kubevirt_migrate_vmi_running_count",
-			Help: "Number of current running migrations.",
-		},
-		migrationsLabels,
-	)
-
-	MigrationsSucceededTotal = prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "kubevirt_migrate_vmi_succeeded_total",
-			Help: "Number of migrations successfully executed.",
-		},
-		migrationsLabels,
-	)
-
-	MigrationsFailedTotal = prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "kubevirt_migrate_vmi_failed_total",
-			Help: "Number of failed migrations.",
-		},
-		migrationsLabels,
-	)
-)
-
 func RegisterMigrationMetrics(vmiMigrationInformer cache.SharedIndexInformer) {
-	log.Log.Infof("Starting migration's counter metrics")
-	prometheus.MustRegister(CurrentPendingMigrations)
-	prometheus.MustRegister(CurrentSchedulingMigrations)
-	prometheus.MustRegister(CurrentRunningMigrations)
-	prometheus.MustRegister(MigrationsSucceededTotal)
-	prometheus.MustRegister(MigrationsFailedTotal)
 	log.Log.Infof("Starting migration's performance and scale metrics")
 	prometheus.MustRegister(newVMIMigrationPhaseTransitionTimeFromCreationHistogramVec(vmiMigrationInformer))
 }
-
-func IncPendingMigrations(vmi *virtv1.VirtualMachineInstance, targetPod *k8sv1.Pod) {
-	updateMigrationGauge(vmi, targetPod, CurrentPendingMigrations, increase)
-}
-
-func IncSchedulingMigrations(vmi *virtv1.VirtualMachineInstance, targetPod *k8sv1.Pod) {
-	updateMigrationGauge(vmi, targetPod, CurrentSchedulingMigrations, increase)
-}
-
-func IncRunningMigrations(vmi *virtv1.VirtualMachineInstance, targetPod *k8sv1.Pod) {
-	updateMigrationGauge(vmi, targetPod, CurrentRunningMigrations, increase)
-}
-
-func DecPendingMigrations(vmi *virtv1.VirtualMachineInstance, targetPod *k8sv1.Pod) {
-	updateMigrationGauge(vmi, targetPod, CurrentPendingMigrations, decrease)
-}
-
-func DecSchedulingMigrations(vmi *virtv1.VirtualMachineInstance, targetPod *k8sv1.Pod) {
-	updateMigrationGauge(vmi, targetPod, CurrentSchedulingMigrations, decrease)
-}
-
-func DecRunningMigrations(vmi *virtv1.VirtualMachineInstance, targetPod *k8sv1.Pod) {
-	updateMigrationGauge(vmi, targetPod, CurrentRunningMigrations, decrease)
-}
-
-func IncSucceededMigrations(vmi *virtv1.VirtualMachineInstance, targetPod *k8sv1.Pod) {
-	incMigrationCounter(vmi, targetPod, MigrationsSucceededTotal)
-}
-
-func IncFailedMigrations(vmi *virtv1.VirtualMachineInstance, targetPod *k8sv1.Pod) {
-	incMigrationCounter(vmi, targetPod, MigrationsFailedTotal)
-}
-
-func getMigrationSourceAndTarget(vmi *virtv1.VirtualMachineInstance, targetPod *k8sv1.Pod) (source, target string) {
-	source = vmi.Status.NodeName
-	if targetPod != nil {
-		target = targetPod.Spec.NodeName
-	}
-
-	if vmi.Status.MigrationState != nil {
```

Comment:
@machadovilaca You could preserve the source and target nodes information in the new logic, by adding a VMI informer, then reading the the VMI status using the VMI name from the VMIM. Is there a reason you decided to drop the source and target node labels?

Reply:
We are not using that information in any alerts or recording rules, and since in this PR I added the VMIM name information in the labels, it's very easy for the user to get the source and target node information starting from the VMIM name